src/sentry/services/hybrid_cloud/notifications/impl.py

```diff
@@ -271,10 +295,20 @@ def apply_filters(
                 query = query.filter(scope_type=filters["scope_type"])
             if "scope_identifier" in filters and filters["scope_identifier"] is not None:
                 query = query.filter(scope_identifier=filters["scope_identifier"])
-            if "user_ids" in filters and len(filters["user_ids"]) > 0:
-                query = query.filter(user_id__in=filters["user_ids"])
-            if "team_ids" in filters and len(filters["team_ids"]) > 0:
-                query = query.filter(team_id__in=filters["team_ids"])
+            if (
+                "user_ids" in filters
+                and "team_ids" in filters
+                and len(filters["user_ids"]) > 0
+                and len(filters["team_ids"]) > 0
+            ):
+                query = query.filter(
+                    Q(user_id__in=filters["user_ids"]) | Q(team_id__in=filters["team_ids"])
+                )
```

Comment:
This matches the behaviour of `def _filter()` as implemented here https://github.com/getsentry/sentry/blob/e8c2b619393f3d7667473ba4f1db373fd65cb514/src/sentry/notifications/manager.py#L306-L310